src/corehost/cli/deps_resolver.cpp

```diff
@@ -119,6 +122,197 @@ void deps_resolver_t::get_dir_assemblies(
 }
 
+bool deps_resolver_t::try_roll_forward(const deps_entry_t& entry,
+    const pal::string_t& probe_dir,
+    pal::string_t* candidate)
+{
+    trace::verbose(_X("Attempting a roll forward for [%s/%s/%s] in [%s]"), entry.library_name.c_str(), entry.library_version.c_str(), entry.relative_path.c_str(), probe_dir.c_str());
+
+    const pal::string_t& lib_ver = entry.library_version;
+
+    fx_ver_t cur_ver(-1, -1, -1);
+    if (!fx_ver_t::parse(lib_ver, &cur_ver, false))
+    {
+        trace::verbose(_X("No roll forward as specified version [%s] could not be parsed"), lib_ver.c_str());
+        return false;
+    }
+
+    // Extract glob string of the form: 1.0.* from the version 1.0.0-prerelease-00001.
+    size_t pat_start = lib_ver.find(_X('.'), lib_ver.find(_X('.')) + 1);
+    pal::string_t maj_min_star = lib_ver.substr(0, pat_start + 1) + _X('*');
+
+    pal::string_t path = probe_dir;
+    append_path(&path, entry.library_name.c_str());
+
+    pal::string_t cache_key = path;
+    append_path(&cache_key, maj_min_star.c_str());
+
+    pal::string_t max_str;
+    if (m_roll_forward_cache.count(cache_key))
+    {
+        max_str = m_roll_forward_cache[cache_key];
+        trace::verbose(_X("Found cached roll forward version [%s] -> [%s]"), lib_ver.c_str(), max_str.c_str());
+    }
+    else
+    {
+        try_patch_roll_forward_in_dir(path, cur_ver, &max_str);
+        m_roll_forward_cache[cache_key] = max_str;
+    }
+
+    append_path(&path, max_str.c_str());
+
+    return entry.to_rel_path(path, candidate);
+}
+
+void deps_resolver_t::setup_probe_config(
+    const corehost_init_t* init,
+    const runtime_config_t& config,
+    const arguments_t& args)
+{
+    if (pal::directory_exists(args.dotnet_extensions))
+    {
+        pal::string_t ext_ni = args.dotnet_extensions;
+        append_path(&ext_ni, get_arch());
+        if (pal::directory_exists(ext_ni))
+        {
+            // Servicing NI probe.
+            m_probes.push_back(probe_config_t::svc_ni(ext_ni, config.get_fx_roll_fwd()));
+        }
+
+        // Servicing normal probe.
+        m_probes.push_back(probe_config_t::svc(args.dotnet_extensions, config.get_fx_roll_fwd()));
+    }
+
+    if (pal::directory_exists(args.dotnet_packages_cache))
+    {
+        pal::string_t ni_packages_cache = args.dotnet_packages_cache;
+        append_path(&ni_packages_cache, get_arch());
+        if (pal::directory_exists(ni_packages_cache))
+        {
+            // Packages cache NI probe
+            m_probes.push_back(probe_config_t::cache_ni(ni_packages_cache));
+        }
+
+        // Packages cache probe
+        m_probes.push_back(probe_config_t::cache(args.dotnet_packages_cache));
+    }
+
+    if (pal::directory_exists(m_fx_dir))
+    {
+        // FX probe
+        m_probes.push_back(probe_config_t::fx(m_fx_dir, m_fx_deps.get()));
+    }
+
+    for (const auto& probe : m_additional_probes)
+    {
+        // Additional paths
+        assert(pal::directory_exists(probe));
```

Comment:
User could specify an incorrect/non-existent path. Instead of the assert, please add a check for path existence, similar to the earlier checks.  Finally, please add an entry to tracelog for all the paths (FX, Dotnetextensions, probepaths, etc) that are found/not-found.